core-backend/src/funcs.rs

```diff
@@ -675,6 +669,30 @@ where
         })
     }
 
+    pub fn free_range(start: u32, end: u32) -> impl Syscall<Ext, i32> {
+        InfallibleSyscall::new(RuntimeCosts::FreeRange, move |ctx: &mut CallerWrap<Ext>| {
+            let page_err = |_| {
+                UndefinedTerminationReason::Actor(ActorTerminationReason::Trap(
+                    TrapExplanation::Unknown,
+                ))
+            };
+
+            let start = WasmPage::new(start).map_err(page_err)?;
+            let end = WasmPage::new(end).map_err(page_err)?;
+
+            match ctx.ext_mut().free_range(start, end) {
```

Comment:
Result should be passed into `process_alloc_func_result` so program is halted during charge error